client/src/actions/index.ts

```diff
@@ -222,18 +240,12 @@ const dispatchDiffExpErrors = (dispatch: any, response: any) => {
 };
 
 const requestDifferentialExpression =
-  (
-    // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any -- - FIXME: disabled temporarily on migrate to TS.
-    set1: any,
-    // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any -- - FIXME: disabled temporarily on migrate to TS.
-    set2: any,
-    num_genes = 50
-    // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any -- - FIXME: disabled temporarily on migrate to TS.
-  ) =>
-  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types, @typescript-eslint/no-explicit-any -- - FIXME: disabled temporarily on migrate to TS.
-  async (dispatch: any, getState: any) => {
+  (set1: LabelArray, set2: LabelArray, num_genes = 50) =>
+  // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types -- TODO: type diff exp data in genesets reducer (client/src/reducers/genesets.ts)
+  async (dispatch: AppDispatch, getState: GetState) => {
     dispatch({ type: "request differential expression started" });
     try {
+      if (!globals.API) throw new Error("API not set");
```

Comment:
Is there a cleaner way of throwing this error? Realistically, if there is no API set, the app is DOA and we have already thrown an error

Reply:
Yeah I think this is fine, since it's unlikely we'll run into this condition IRL, but defensive programming is great. So keeping it sounds good!